packages/react/src/react/ssr.tsx

```diff
@@ -27,14 +26,60 @@ interface CreateComponentForServerSideRenderingOptions {
 
 type StencilProps<I extends HTMLElement> = WebComponentProps<I>;
 
+type LazyComponent<T, P> = {
+  $$typeof: symbol | number;
+  _payload: P;
+  _init: (payload: P) => T;
+};
+
+type ReactNodeExtended = ReactNode | Component<any, any, any> | LazyComponent<any, any>;
+
 /**
  * returns true if the value is a primitive, e.g. string, number, boolean
  * @param value - the value to check
  * @returns true if the value is a primitive, false otherwise
  */
-function isPrimitive(value: any) {
-  return typeof value === 'string' || typeof value === 'number' || typeof value === 'boolean';
-}
+const isPrimitive = (value: any): value is string | number | boolean =>
+  typeof value === 'string' || typeof value === 'number' || typeof value === 'boolean';
+
+/**
+ * returns true if the value is empty, e.g. null or undefined
+ * @param value - the value to check
+ * @returns true if the value is empty, false otherwise
+ */
+const isEmpty = (value: any): value is null | undefined => value === null || value === undefined;
```

Comment:
Wouldn't it be better to type the input value as `unknown`, same with other checks further below?

```suggestion
const isEmpty = (value: unknown): value is null | undefined => value === null || value === undefined;
```